packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts

```diff
@@ -22,24 +23,41 @@ export const useVisualizedDataStreams = (queries: TimeSeriesDataQuery[], passedI
     },
   });
 
-  // Line | Scatter | Bar charts do not support alarm streams.
-  const dataStreamsWithoutAlarms = dataStreams.filter(isNotAlarmStream);
-
-  const hasError = useMemo(() => dataStreamsWithoutAlarms.some(dataStreamHasError), [dataStreamsWithoutAlarms]);
-
-  const isLoading = useMemo(
-    () => !hasError && dataStreamsWithoutAlarms.some(dataStreamIsLoading),
-    [hasError, dataStreamsWithoutAlarms]
-  );
-
-  const visibleData = dataStreamsWithoutAlarms.flatMap(({ data }) => getVisibleData(data, utilizedViewport, false));
+  // dependencies
+  const dataString = JSON.stringify(dataStreams);
+  const thresholdsString = JSON.stringify(thresholds);
+  const utilizedString = JSON.stringify(utilizedViewport);
+
+  // utilizing state to make sure the values are updated only on change
+  const [dataStreamsWithoutAlarms, setDataStreamsWithoutAlarms] = useState(dataStreams);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [performanceMode, setPerformanceMode] = useState(false);
```

Comment:
What does performanceMode do? Are there comments explaining what it optimizes?

Reply:
this is an existing logic from Jason, we were exporting an array from this hook which was only used in calculating this boolean variable called performanceMode... which is nothing by # of visible elements > 4000, so i moved this inside this hook and just exporting the boolean instead